diff --git a/lib/core/connect.js b/lib/core/connect.js
--- a/lib/core/connect.js
+++ b/lib/core/connect.js
@@ -71,7 +71,7 @@ function buildConnector ({ allowH2, useH2c, maxCachedSessions, socketPath, timeo
         highWaterMark: 16384, // TLS in node can't have bigger HWM anyway...
         ...options,
         servername,
-        session,
+        ...(session != null ? { session } : {}),
         localAddress,
         ALPNProtocols: allowH2 ? ['http/1.1', 'h2'] : ['http/1.1'],
         socket: httpSocket, // upgrade socket connection
